lrlex/src/lib/parser.rs

```diff
@@ -201,20 +200,22 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
         if declaration_parameters.is_empty() {
             return Err(self.mk_error(LexErrorKind::UnknownDeclaration, off));
         }
-        let start_states: HashSet<(&str, Span)> = declaration_parameters
+        let start_states: Vec<(&str, Span)> = declaration_parameters
             .split_whitespace()
             .map(|name| {
                 let off = name.as_ptr() as usize - self.src.as_ptr() as usize;
                 let span = Span::new(off, off + name.len());
-                self.validate_start_state(span, name, errs)
+                (name, span)
             })
-            .filter_map(Result::transpose)
-            .collect::<LexInternalBuildResult<HashSet<(&str, Span)>>>()?;
+            .collect::<Vec<(&str, Span)>>();
```

Comment:
Do you really want to *both* explicitly type in the variable `start_states`' declaration, and turbo-fish in the `collect`?

Reply:
I'd failed to notice because of inline hints, we could even just simplify it to just `Vec<_>` in collect.

Reply:
Should be fixed in [658db78](https://github.com/softdevteam/grmtools/pull/332/commits/658db78e67fd8c0ebb1df5a5dd2c27535bd182ef)